### PR TITLE
#4217 add session storage for wallet connect

### PR DESCRIPTION
**Pull request by desfero**
**TODO:**
- [x] Check if there is a memory leak after the session was disconnected
- [x] Test whether sign message is working after @rudolfix fixes web app
- [ ] Clean up session when the new user was signed

### CODE REVIEW

File: packages/wallet/app/components/shared/Link.tsx [hunk 1]
```diff
@@ -2,74 +2,49 @@
...
-  const onShowUnderlay = () => {
-    setIsActive(true);
-  };
-
-  const onHideUnderlay = () => {
-    setIsActive(false);
-  };
-
-  return (
-    <TouchableHighlight
```
Comment (desfero) [inline packages/wallet/app/components/shared/Link.tsx:44]: I wasn't aware that it's not possible to nest `TouchableHighligh` into the `Text` in react-native. But that's good as we can simplify the implementation to only have a `Text` node here

File: packages/wallet/app/modules/init/sagas.ts [hunk 2]
```diff
@@ -31,6 +33,11 @@ function* initStartSaga(): SagaGenerator<void> {
   }
 }
 
+function* initAuthSigned(): SagaGenerator<void> {
+  yield* call(walletConnectModuleApi.sagas.tryToConnectExistingSession);
+}
+
 export function* initSaga(): SagaGenerator<void> {
   yield fork(neuTakeLatest, initActions.start, initStartSaga);
+  yield fork(neuTakeLatest, authActions.signed, initAuthSigned);
```
Comment (desfero) [inline packages/wallet/app/modules/init/sagas.ts:42]: not sure if it's the best approach here as now init module watches for `authActions.signed`. In general, it looks good to me as we avoid coupling `auth` module with `wallet-connect` module given that they are not aware of each other and it's part of `init` module to connect them properly

File: packages/wallet/app/modules/wallet-connect/lib/WalletConnectAdapter.ts [hunk 3]
```diff
@@ -0,0 +1,300 @@
+import { ILogger } from "@neufund/shared-modules";
```
Comment (desfero) [inline packages/wallet/app/modules/wallet-connect/lib/WalletConnectAdapter.ts:1]: almost a one to one copy from `WalletConnectManager` given the manager was repurposed to control session storage and instantiation of WalletConnectAdapter

File: packages/wallet/app/modules/wallet-connect/lib/WalletConnectAdapter.ts [hunk 4]
```diff
@@ -0,0 +1,300 @@
...
+ * Wraps wallet connect class to simplify interface and to provide type safety
+ *
+ * @internal WalletConnectAdapter should only be used in WalletConnectManager and never exposed to DI container
+ */
+class WalletConnectAdapter extends EventEmitter2 {
+  private readonly logger: ILogger;
+
+  private readonly walletConnect: WalletConnect;
+
+  private connectedAt: number | undefined = undefined;
```
Comment (desfero) [inline packages/wallet/app/modules/wallet-connect/lib/WalletConnectAdapter.ts:72]: `connectedAt` was added to the adapter in this PR

**Comment by arjunu** [comment]
Closed in favor of https://github.com/Neufund/platform-frontend/pull/4362

**Pull request closed without merging**